velox/connectors/Connector.h

```diff
@@ -304,7 +305,8 @@ class Connector {
       const std::unordered_map<
           std::string,
           std::shared_ptr<connector::ColumnHandle>>& columnHandles,
-      ConnectorQueryCtx* FOLLY_NONNULL connectorQueryCtx) = 0;
+      ConnectorQueryCtx* FOLLY_NONNULL connectorQueryCtx,
+      const std::shared_ptr<dwio::common::ReaderOptions>& options = {}) = 0;
```

Comment:
`const std::shared_ptr<dwio::common::ReaderOptions> options = {}` is cleaner (no reference) here and elsewhere. We can use `std::move` to optimize.